modules/antismash/antismashlite/main.nf

```diff
@@ -0,0 +1,143 @@
+//     tuple val(meta), path("${prefix}/*.json")                                , emit: json_results
+//     tuple val(meta), path("${prefix}/*.log")                                 , emit: log
+//     tuple val(meta), path("${prefix}/*.zip")                                 , emit: zip
+//     tuple val(meta), path("${prefix}/*region*.gbk")                          , emit: gbk_results
+//     tuple val(meta), path("${prefix}/clusterblastoutput.txt")                , optional: true, emit: clusterblastoutput
+//     tuple val(meta), path("${prefix}/index.html")                            , emit: html
+//     tuple val(meta), path("${prefix}/knownclusterblastoutput.txt")           , optional: true, emit: knownclusterblastoutput
+//     tuple val(meta), path("${prefix}/regions.js")                            , emit: json_sideloading
+//     path "versions.yml"                                                      , emit: versions
+
+//     when:
+//     task.ext.when == null || task.ext.when
+
+//     script:
+//     def args = task.ext.args ?: ''
+//     prefix = task.ext.suffix ? "${meta.id}${task.ext.suffix}" : "${meta.id}"
+//     // if ( sequence_input.getExtension != 'fasta' && sequence_input.getExtension != 'fna' && gff )
+//     //     log.warn "GFF input to antiSMASH can only be used if FASTA sequence input is supplied. GFF will be ignored for sample ${meta.id}"
+//     // if ( (sequence_input.getExtension == 'fasta' || sequence_input.getExtension == 'fna') && gff )
+//     //     gff_flag = "--genefinding-gff3 ${gff}"
+//     // else
+//     //     gff_flag = ""
+
+//     """
+//     ## We specifically do not include annotations (--genefinding-tool none) as
+//     ## this should be run as a separate module for versioning purposes
+//     antismash \\
+//         $args \\
+//         -c $task.cpus \\
+//         --output-dir $prefix \\
+//         --genefinding-tool none \\
+//         --logfile $prefix/${prefix}.log \\
+//         --databases $databases \\
+//         $sequence_input
+
+//     cat <<-END_VERSIONS > versions.yml
+//     "${task.process}":
+//         antismash-lite: \$(antismash --version | sed 's/antiSMASH //')
+//     END_VERSIONS
+//     """
+// }
+process ANTISMASH_ANTISMASHLITE {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::antismash-lite=6.0.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/antismash-lite:6.0.1--pyhdfd78af_1' :
+        'quay.io/biocontainers/antismash-lite:6.0.1--pyhdfd78af_1' }"
+
+    containerOptions {
+        workflow.containerEngine == 'singularity' ?
+        "-B $antismash_dir:/usr/local/lib/python3.8/site-packages/antismash" :
+        workflow.containerEngine == 'docker' ?
+        "-v \$PWD/$antismash_dir:/usr/local/lib/python3.8/site-packages/antismash" :
```

Comment:
Is there not a sub directory of antismash where this needs to be placed? You are replacing the entire antismash Python package here (it seems to me) which undermines the code stability and reproducibility otherwise guaranteed by using a container image with pre-installed software.

Reply:
That's exactly the tricky point – as described in the helper module "antismashlite-downloaddatabases" in [this comment](https://github.com/nf-core/modules/blob/9602b989ebb30b68196f53720aa388952027d460/modules/antismash/antismashlitedownloaddatabases/main.nf#L12), we need to mount the antismash installation directory to the container image because it is being modified during the database download. You would need root permissions in the container to handle modifications of the installation directory (`/usr/local/lib/python3.8/site-packages/antismash/`). Mounting is the only workaround that we see here. Which unfortunately renders the whole image a bit useless :confused: 
Conda is totally fine, though :grimacing: